### DISCUSSION TITLE
Redirection outside component

### DISCUSSION THREAD

**Discussion by fab918** [molefrog/wouter#385] [Q&A]
I'd like to replace react router with Wouter, especially because navigate out of the component is a pain, which is useful/needed with applications using Redux/Saga for redirects.

What is the recommended method with Wouter ?

Since Wouter uses the History API, is there a problem with using the History API directly in these cases?

Something like that?
``
export const navigate = (to) => {
  window.history.pushState(null, '', to);
  window.dispatchEvent(new PopStateEvent('popstate'));
};
``

If not, it think it will be interesting to add few line about it, since in react router it changes often, it breaks and nothing official about it.

**Comment by molefrog** [comment]
Hey! First of all, thank you for choosing wouter. 
I tried providing the solution here https://github.com/molefrog/wouter/issues/59#issuecomment-1810806990

This is essentially the same method wouter returns as a second element from `useLocation()`
